### PR TITLE
Added support for timestamp.Timestamp

### PR DESCRIPTION
**Pull request by lucasvo**

### CODE REVIEW

File: proofs/tree.go [hunk 1]
```diff
@@ -185,6 +194,9 @@ func ValueToString(value interface{}) (s string, err error) {
 		return strconv.FormatInt(value.(int64), 10), nil
 	case reflect.TypeOf([]uint8{}):
 		return base64.StdEncoding.EncodeToString(value.([]uint8)), nil
+	case reflect.TypeOf(&timestamp.Timestamp{}):
+		v := reflect.ValueOf(value).Elem().Interface().(timestamp.Timestamp)
```
Comment (mikiquantum) [inline proofs/tree.go]: I believe `reflect.ValueOf(value)` already returns `Value` type. I do not think `Elem()` is needed.

Comment (lucasvo) [reply proofs/tree.go]: the value is actually of type *timestamp.Timestmap (i.e. it is a pointer, therefore I need to dereference it with Elem()

Comment (mikiquantum) [reply proofs/tree.go]: Ah ok. I didn't realized that TimestampProto returns a pointer. Misleading as the other supported types are passed by value. On the other hand, timestamp is the first that is a more "complex" struct, and usually structs are passed by reference. All good.

Comment (mikiquantum) [reply proofs/tree.go]: Although, thinking about it a bit more, as we are not in control on how the timestamp is defined in the complex struct to be parsed, I would add another case where we support the dereferenced value as well. Like:
```
case reflect.TypeOf(timestamp.Timestamp{}):
  v := reflect.ValueOf(value).Interface().(timestamp.Timestamp)
```
Or even better, enforcing dereferencing when passing the value to this function.

**Pull request merged**